### ISSUE TITLE
http.Server doesn't expose randomly selected port

### ISSUE THREAD

**Issue by aricart** [denoland/deno_std#1036]
If you specify port `0`, as `serve({ port: 0 })` the library works correctly and automatically selects a port. However, the port is accessible to the process cannot advertise where it is listening on. In some cases it is desirable to start a process on any available port, and then advertise it, be it through the console, or some other transport/file.

**Comment by getspooky** [comment]
Fixed https://github.com/denoland/deno_std/pull/1037

**Comment by getspooky** [comment]
Hi @aricart you can simply run `sudo lsof -i -P -n | grep LISTEN` to get port 
```sh
deno      17014    12u  IPv4 131998      0t0  TCP *:43523 (LISTEN)
```

**Comment by AaronO** [comment]
@aricart @getspooky You can actually already to this, though it's somewhat quirky (`.port` and `Addr` typing), see https://github.com/denoland/deno/pull/11410#issuecomment-880747416:

```ts
import { serve } from "https://deno.land/std@0.101.0/http/server.ts";
const server = serve({ port: 0 });
const serverAddr = server.listener.addr as Deno.NetAddr;
console.log(`http://localhost:${serverAddr.port}`);
for await (const req of server) {
  req.respond({ body: "Hello World\n" });
}
```

**Comment by aricart** [comment]
> Hi @aricart you can simply run `sudo lsof -i -P -n | grep LISTEN` to get port
> 
> ```shell
> deno      17014    12u  IPv4 131998      0t0  TCP *:43523 (LISTEN)
> ```
> 
> But i think deno should provide that by default.

But that doesn't do it -  What if I am running 100 deno processes? - The app should be able to (trivially) expose it.

**Comment by getspooky** [comment]
> > Hi @aricart you can simply run `sudo lsof -i -P -n | grep LISTEN` to get port
> > ```shell
> > deno      17014    12u  IPv4 131998      0t0  TCP *:43523 (LISTEN)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > But i think deno should provide that by default.
> 
> But that doesn't do it - What if I am running 100 deno processes? - The app should be able to (trivially) expose it.

Yes , I agree with you, so you can use AaronO proposition it's the best one.

**Comment by iuioiua** [comment]
The current API automatically prints where the server is listening. It can also be manually controlled via the `onListen` property, if needed. For example:
```ts
import { serve } from "https://deno.land/std/http/server.ts";

serve(() => new Response("Hello, world!"), { port: 0 });
serve(() => new Response("Hello, world!"), { port: 0 });
serve(() => new Response("Hello, world!"), { port: 0 });
```
@kt3k, I believe this issue can be closed.

**Comment by kt3k** [comment]
@iuioiua Thanks. Right, this was resolved in #2311. Now the listening port is available in the option of `onListen` handler.
